src/main/java/org/sunbird/digilocker/service/DigiLockerIntegrationServiceImpl.java

```diff
@@ -0,0 +1,243 @@
+        try (ByteArrayOutputStream outputStream = new ByteArrayOutputStream()) {
+            TranscoderInput input = new TranscoderInput(new ByteArrayInputStream(svgContentDecode.getBytes()));
+            TranscoderOutput output = new TranscoderOutput(outputStream);
+            PDFTranscoder transcoder = new PDFTranscoder();
+            transcoder.transcode(input, output);
+            return outputStream.toByteArray();
+        }
+    }
+
+    private void encodeToFile(String encodedString) {
+
+        try {
+            byte[] decodedBytes = Base64.getDecoder().decode(encodedString);
+            String filePath = "/home/sahilchaudhary/Desktop/files.pdf";
+
+            // Write the decoded data to a file
+            try (FileOutputStream fos = new FileOutputStream(filePath)) {
+                fos.write(decodedBytes);
+            }
+
+            System.out.println("Decoded data saved to: " + filePath);
+        } catch (IOException e) {
+            System.err.println("Error: " + e.getMessage());
+        }
+    }
+
+    private static String decodeUrl(String encodedUrl) throws UnsupportedEncodingException {
+        StringBuilder decodedUrl = new StringBuilder();
+        int index = 0;
+        while (index < encodedUrl.length()) {
+            char currentChar = encodedUrl.charAt(index);
+            if (currentChar == '%') {
+                String hex = encodedUrl.substring(index + 1, index + 3);
+                decodedUrl.append((char) Integer.parseInt(hex, 16));
+                index += 3;
+            } else {
+                decodedUrl.append(currentChar);
+                index++;
+            }
+        }
+        return decodedUrl.toString();
+    }
+
+
+    private boolean addUpdateDigiLockerLookup(Map<String, Object> userDocInfo) {
+        Map<String, Object> digiLockerInfoMap = getDigiLockerDocInfo((String) userDocInfo.get(Constants.DOC_ID), (String) userDocInfo.get(Constants.DIGI_LOCKER_ID));
+        boolean isUpdatedDigiLockerTable = false;
+        if (MapUtils.isEmpty(digiLockerInfoMap)) {
+            SBApiResponse resp = cassandraOperation.insertRecord(Constants.KEYSPACE_SUNBIRD,
+                    Constants.TABLE_DIGILOCKER_DOC_INFO, userDocInfo);
+            if (!resp.get(Constants.RESPONSE).equals(Constants.SUCCESS)) {
+                isUpdatedDigiLockerTable = false;
+            } else {
+                isUpdatedDigiLockerTable = true;
+            }
+        } else {
+            isUpdatedDigiLockerTable = true;
+        }
+        return isUpdatedDigiLockerTable;
+    }
+
+    private Map<String, Object> getDigiLockerDocInfo(String docId, String digiLockerId) {
+        Map<String, Object> digiLockerInfo = new HashMap<>();
+        digiLockerInfo.put(Constants.DOC_ID, docId);
+        digiLockerInfo.put(Constants.DIGI_LOCKER_ID, digiLockerId);
+        List<Map<String, Object>> digiLockerInfoMap = cassandraOperation.getRecordsByProperties(Constants.KEYSPACE_SUNBIRD,
```

Comment:
getRecordsByProperties -- is it using "allow filtering" ??
if so -- use the proper method which doesn't have allow filtering

Reply:
Used getRecordsByPropertiesWithoutFiltering